README.md

```diff
@@ -109,38 +109,38 @@ Real-world usage examples from code using ICU’s `UnicodeSet` which implements
 - All BMP code points that are either a letter, a mark (e.g. diacritic), or a decimal number:
 
     ```
-    [[:BMP:]&[[:L:][:M:][:Nd:]]]
+    [\p{BMP}&&[\p{L}\p{M}\p{Nd}]]
```

Comment:
For the other existing examples, we add a note if they use a feature that doesn't yet exist in ECMAScript:

```
Note that ECMAScript currently doesn’t support `\p{BMP}` — this is an illustrative example regardless.
```

Let's also use the more readable long-form category names, e.g. `\p{Letter}` instead of `\p{L}`, again to match the other examples. I’d be happy to do all this as a fix-up though, to not block this PR further.